code/modules/admin/sql_notes.dm

```diff
@@ -0,0 +1,214 @@
+/proc/add_note(target_ckey, notetext, timestamp, adminckey, logged = 1, server)
+	if(!dbcon.IsConnected())
+		usr << "<span class='danger'>Failed to establish database connection.</span>"
+		return
+	if(!target_ckey)
+		var/new_ckey = ckey(input(usr,"Who would you like to add a note for?","Enter a ckey",null) as text|null)
+		if(!new_ckey)
+			return
+		var/DBQuery/query_find_ckey = dbcon.NewQuery("SELECT ckey FROM [format_table_name("player")] WHERE ckey = '[new_ckey]'")
+		if(!query_find_ckey.Execute())
+			var/err = query_find_ckey.ErrorMsg()
+			log_game("SQL ERROR obtaining ckey from notes table. Error : \[[err]\]\n")
+			return
+		if(!query_find_ckey.NextRow())
+			usr << "<font color='red'>[new_ckey] has not been seen before, you can only add notes to known players.</font>"
+			return
+		else
+			target_ckey = new_ckey
+	var/target_sql_ckey = sanitizeSQL(target_ckey)
+	if(!notetext)
+		notetext = input(usr,"Write your Note","Add Note") as message
+		if(!notetext)
+			return
+	notetext = sanitizeSQL(notetext)
+	if(!timestamp)
+		timestamp = SQLtime()
+	if(!adminckey)
+		adminckey = usr.ckey
+		if(!adminckey)
+			return
+	var/admin_sql_ckey = sanitizeSQL(adminckey)
+	if(!server)
+		if (config && config.server_name)
+			server = config.server_name
+	var/DBQuery/query_noteadd = dbcon.NewQuery("INSERT INTO [format_table_name("notes")] (ckey, timestamp, notetext, adminckey, server) VALUES ('[target_sql_ckey]', '[timestamp]', '[notetext]', '[admin_sql_ckey]', '[server]')")
+	if(!query_noteadd.Execute())
+		var/err = query_noteadd.ErrorMsg()
+		log_game("SQL ERROR adding new note to table. Error : \[[err]\]\n")
+		return
+	if(logged)
+		log_admin("[key_name(usr)] has added a note to [target_ckey]: [notetext]")
+		message_admins("[key_name_admin(usr)] has added a note to [target_ckey]:<br>[notetext]")
+		show_note(target_ckey)
+
+/proc/remove_note(note_id)
+	var/ckey
+	var/notetext
+	var/adminckey
+	if(!dbcon.IsConnected())
+		usr << "<span class='danger'>Failed to establish database connection.</span>"
+		return
+	if(!note_id)
+		return
+	var/DBQuery/query_find_note_del = dbcon.NewQuery("SELECT ckey, notetext, adminckey FROM [format_table_name("notes")] WHERE id = '[note_id]'")
```

Comment:
ID is a number, and should not have quotes.

You also also check that the note_id var is a number, to prevent sql injection exploits